src/BoundaryConditions/update_boundary_conditions.jl

```diff
@@ -0,0 +1,16 @@
+using Oceananigans: boundary_conditions
+
+update_boundary_condition!(bc::Union{BoundaryCondition, Nothing}, args...) = nothing
+
+function update_boundary_condition!(bcs::FieldBoundaryConditions, field, model)
+    update_boundary_condition!(bcs.west, field, model, Val(:west))
```

Comment:
The motivation for this was when I wrote a mean outflow boundary condition which I accidentally put in this old commit https://github.com/CliMA/Oceananigans.jl/blob/4bf02522aa63679a03c3b32d7e2ee9517fc37b3c/src/BoundaryConditions/open_boundary_matching_schemes.jl

Because then its easier to write generic function to update and dispatch the bits that are actually needed

Reply:
Where is the generic version of `update_boundary_condition!` for a single boundary defined? In this code you've passed in all 7 boundary names that I am aware of.

Reply:
https://github.com/CliMA/Oceananigans.jl/blob/4bf02522aa63679a03c3b32d7e2ee9517fc37b3c/src/BoundaryConditions/open_boundary_matching_schemes.jl#L39-L51

This bit and then the function inside dispatch on `side`

Reply:
I see. Ok, well then my only comment is to fix the order of the function arguments. The first argument should be the object that is mutated (if any), followed by arguments that are dispatched against, followed by the rest.

Reply:
By the way, is this function correct? It divides by `Ly * Lz` which seems to apply only to east and west

Reply:
and the kernel launches over `:yz`

Reply:
Ah okay so it should be `bc, side, field, model` in this case then right?

Reply:
I think so, because it updates `bc.classification.matching_scheme.boundary_flux`?

Reply:
confused about correctness though. But this file wasn't touched by this PR is that right? I can't find it in the diff

Reply:
also `u = boundary_normal_velocity(model.velocities, side) ` should be `u = boundary_normal_velocity(side, model.velocities)

Reply:
yeah that matching scheme I ended up not including in this PR but I accidentally pushed it earlier

Reply:
where is `update_boundary_condition!` in the current code?

Reply:
This is the fallback https://github.com/CliMA/Oceananigans.jl/blob/cd218e1e3e1d3cb9af5bbfa44a4445e9dd8f58c1/src/BoundaryConditions/update_boundary_conditions.jl#L3

Its not currently used for anything.